### PR TITLE
Replace direct usage of ACTIVE_WORLD_OPTIONS with calendar::season_length

### PR DESCRIPTION
**Pull request by BevapDin**
The calendar function uses the cached value and does not need checking for ACTIVE_WORLD_OPTIONS being empty.

This also allows to get rid of the include of worldfactory.h in iexamine.cpp as it is no longer needed there.

Edit: it also removes `calendar::set_season_length`, `calendar::season_length` gets its values directly from the world options.

And I moved setting the changed changed world options (`ACTIVE_WORLD_OPTIONS`, edited in `show`) to the active `WORLD` (where they are stored) to `show`. See discussion and commit messages.

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -14561,11 +14561,11 @@ void game::start_calendar()
         if( scen->has_flag("SPR_START") ) {
             ; // Do nothing;
         } else if( scen->has_flag("SUM_START") ) {
-            calendar::start += DAYS((int)ACTIVE_WORLD_OPTIONS["SEASON_LENGTH"]);
+            calendar::start += DAYS( calendar::season_length() );
```
Comment (codemime) [inline src/game.cpp]: These changes (x6) don't look consistent. The function's name is `start_calendar()` so it's supposed to initialize it, not rely on it's values. There are alot of `ACTIVE_WORLD_OPTIONS` around whilst `calendar::set_season_length` is invoked from fairly weird places (`options_manager::serialize` and `worldfactory::set_active_world`). It should probably be placed here and be initialized from `ACTIVE_WORLD_OPTIONS` too.

Comment (BevapDin) [inline src/game.cpp]: > whilst  calendar::set_season_length  is invoked from fairly weird places ( options_manager::serialize  and  worldfactory::set_active_world ).

It gets executed when the contents of `ACTIVE_WORLD_OPTIONS` changes. Those are exactly the places where it must be.

> It should probably be placed here and be initialized from  ACTIVE_WORLD_OPTIONS  too.

It needs to be set when the player changes the world options while already playing the game, therefor in `options_manager::serialize` (called from `options_manager::show`).

Through it should probably be set directly in `options_manager::show` - but that makes not actual difference for this PR.

> These changes (x6) don't look consistent. The function's name is  start_calendar()  so it's supposed to initialize it, not rely on its values.

The function "starts" `calendar::start` and `calendar::turn` (actual `calendar` objects). `calendar::season_length` is not really part of them, it's a static function, accessing global data.

For what it's worth, `calendar::season_length` could as well be a global function, or a game function or whatever. 

Actually `calendar::turn` and `calendar::start` should probably be moved to `game`.

----

`calendar::season_length` is a simple wrapper around `ACTIVE_WORLD_OPTIONS["SEASON_LENGTH"]`, nothing more. The PR is supposed to replace the separate ways of accessing the option value (via `ACTIVE_WORLD_OPTIONS` and via `season_length`) with a single (established) way.

You can see it like that: where do you look if you want to know the season length? The options or the calendar? How do you even know it's an option?

Comment (codemime) [inline src/game.cpp]: > Those are exactly the places where it must be.

I disagree. These places are WTFs at best. `options_manager::serialize` is supposed to serialize the content.  And nothing more. `worldfactory::set_active_world` is a bit lesser WTF, but it doesn't change the fact. Setting season length is a side effect of these functions which pollutes them. In other words, it's a code smell.

> The PR is supposed to replace the separate ways of accessing the option value (via ACTIVE_WORLD_OPTIONS and via season_length) with a single (established) way.

Two completely identical calls of `calendar::set_season_length( ACTIVE_WORLD_OPTIONS["SEASON_LENGTH"] );` from two fairly unexpected places is the exact reason why it fails to do so.

Comment (BevapDin) [inline src/game.cpp]: I removed `set_season_length`, `season_length` now gets its values directly from the world options.

**Pull request merged**